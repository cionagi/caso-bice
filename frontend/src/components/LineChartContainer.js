import React from 'react';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  Tooltip,
  CartesianGrid,
  ResponsiveContainer,
} from 'recharts';
import PropTypes from 'prop-types';

const LineChartContainer = ({ historicData }) => {
  return (
    <ResponsiveContainer width="99%" aspect={3}>
      <LineChart
        width={800}
        height={400}
        data={historicData}
        margin={{ top: 40, right: 40, bottom: 20, left: 20 }}
      >
        <CartesianGrid vertical={false} />
        <XAxis dataKey="date" />
        <YAxis dataKey="value" domain={['auto', 'auto']} />
        <Tooltip />
        <Line dataKey="value" label="Valor" stroke="#ff7300" dot={false} />
      </LineChart>
    </ResponsiveContainer>
  );
};

LineChartContainer.propTypes = {
  historicData: PropTypes.array,
};

export default LineChartContainer;
