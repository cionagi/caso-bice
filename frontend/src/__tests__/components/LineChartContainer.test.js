import React from 'react';
import { render } from '@testing-library/react';

import LineChartContainer from '../../components/LineChartContainer';

test('Link renders correctly', () => {
  const { asFragment } = render(<LineChartContainer />);
  expect(asFragment()).toMatchSnapshot();
});
