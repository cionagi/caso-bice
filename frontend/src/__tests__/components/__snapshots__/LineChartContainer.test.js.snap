// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <div
    class="recharts-responsive-container"
    style="width: 99%; height: 100%;"
  >
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</DocumentFragment>
`;
